### PR TITLE
Add a testing section to the documentation

### PR DESCRIPTION
**Pull request by supreeth7**
### What type of PR is this?

documentation

### What this PR does / Why we need it?

Adds a new Testing section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,4 +27,16 @@ Initially, PKO package deploys 4 resources:
...
+
+## Testing
+
+The [Package Operator](https://github.com/package-operator/package-operator) includes a template test framework that makes it easier to identify template errors without having the need of deploying the Package.
+
+For each template test, Package Operator will auto-generate fixtures into the `.test-fixtures` and the test cases that can be defined in the PackageManifest.
+
+To validate the package run this command in the project's root:
+```
+$ kubectl package validate ./package
```
Comment (bmeng) [inline README.md:40]: Do we need some prerequisites here? like install the package kubectl plugin.

And seems it failed in my local testing, not sure if the version of the package is too old
```
$ kubectl package validate ./package
Error: validate package and files: parsing template from resources.yaml.gotmpl: template: :133: function "splitn" not defined
```

Comment (supreeth7) [reply README.md:40]: We need the `kubectl package` plugin: https://github.com/package-operator/package-operator/releases/tag/v1.7.0

Comment (bmeng) [reply README.md:45]: thanks! it works well after I updated my package plugin to latest.

**Pull request merged**